genesyscloud/outbound_campaign/resource_genesyscloud_outbound_campaign.go

```diff
@@ -180,38 +180,28 @@ func updateOutboundCampaign(ctx context.Context, d *schema.ResourceData, meta in
 
 // deleteOutboundCampaign is used by the outbound_campaign resource to delete an outbound campaign from Genesys cloud
 func deleteOutboundCampaign(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
-	sdkConfig := meta.(*gcloud.ProviderMeta).ClientConfig
-	proxy := newOutboundCampaignProxy(sdkConfig)
+	clientConfig := meta.(*gcloud.ProviderMeta).ClientConfig
+	proxy := getOutboundCampaignProxy(clientConfig)
 
 	campaignStatus := d.Get("campaign_status").(string)
 
 	// Campaigns have to be turned off before they can be deleted
 	if campaignStatus != "off" {
-		diagErr := gcloud.RetryWhen(gcloud.IsStatus400, func() (*platformclientv2.APIResponse, diag.Diagnostics) {
-			log.Printf("Turning off Outbound Campaign before deletion")
-			outboundCampaign, resp, getErr := proxy.getOutboundCampaignById(ctx, d.Id())
-			if getErr != nil {
-				return resp, diag.Errorf("Failed to read Outbound Campaign %s: %s", d.Id(), getErr)
-			}
-			// Handles updating the campaign based on what is set in ResourceData.campaign_status
-			diagErr := updateOutboundCampaignStatus(ctx, d.Id(), proxy, *outboundCampaign, "off")
-			if diagErr != nil {
-				return resp, diagErr
-			}
-			return resp, nil
-		})
-		if diagErr != nil {
+		log.Printf("Turning off Outbound Campaign before deletion")
```

Comment:
I refactored the above code into the proxy to reduce function size. Also, the code above was making the mistake of retrying on 400, where really we should be updating and then retrying until the field CampaignStatus != "off" . The 400 appeared when we tried to delete it while it was still "on"